### PR TITLE
tv3g3 php8 errors with Bulk update all titles

### PR DESCRIPTION
**Pull request by dsenalik**
# Bug Fix

## Description
The changes resolve warnings that show up under php8.x, where a job is passed to a callback that does not expect an additional (unnamed) parameter

## Testing?
1. Need to run under PHP8.x
2. Go to organism content type /admin/structure/bio_data/manage/bio_data_1 (or any other content type)
3. Click the "Bulk update all titles" button
4. run the job with drush at the command line and hopefully no errors now appear, see issue #1401 for the "before" output.
```
2023-01-31 13:38:56: There are 1 jobs queued.
2023-01-31 13:38:56: Job ID 3977.
2023-01-31 13:38:56: Calling: tripal_update_all_urls_and_titles(bio_data_4, study/[TripalEntity__entity_id], alias)
1/1 entities have been updated. Memory usage: 52,179,464 bytes.
Done.
```

### CODE REVIEW

**Review by laceysanderson — APPROVED** [review]
Code-wise, this looks great 👍 Thanks @dsenalik 

Note: I don't have an environment up to quickly test this so I have done a code review only.

**Comment by dsenalik** [comment]
This has a side effect, other callbacks will need to be updated
```
WD php: Error: Unknown named parameter $job in TripalJob->run() (line 342 of   [error]
.../sites/all/modules/tripal/tripal/includes/TripalJob.inc).
```
in this case the callback="chado_populate_mview"
and args=
```
array(2) {
  [0]=>
  string(2) "25"
  ["job"]=>
  object(TripalJob)#446 (8) {
    ["job_id":protected]=>
    NULL
    ["job":protected]=>
    object(stdClass)#451 (20) {
... etc.
```

**Comment by dsenalik** [comment]
If I revert ```$arguments['job'] = $this;``` back to the original ```$arguments[] = $this;``` then this error goes away.

**Comment by dsenalik** [comment]
Because there is no way I could ever know all of the functions that may be callbacks, and I assume other modules could use this too, I really can't change all callback parameters, it is best to do no harm by adding new stuff.
The only functional change is to now only pass the job object to the callback when the last parameter of the callback is ```$job```.
Because of this I removed the additional ```$job``` parameter to ```tripal_update_all_urls_and_titles()``` that I had added, because this change _should_ ensure that it is no longer passed to this callback or any callback not expecting it.

**Comment by laceysanderson** [comment]
I agree with the new approach @dsenalik and your implementation of it looks good. I dismissed my code review from above and and we should probably wait for a functional review on this one. Thanks for being extra diligent -as always ❤️

**Review by spficklin — APPROVED** [review]

**Comment by spficklin** [comment]
Thankd @dsenalik

**Pull request merged**
